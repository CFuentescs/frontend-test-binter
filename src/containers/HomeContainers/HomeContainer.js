import {connect} from 'react-redux'
import HomeComponents from '../../components/Home/HomeComponents'
import * as HomeModule from '../../modules/HomeModule/homeModule'

const mapStateToProps =(state) => ({
    isProcess: HomeModule.getProcess(state),
    isLoading: HomeModule.getLoading(state),
    isError: HomeModule.getError(state),
    isMessageInfo: HomeModule.getMessageInfo(state),
    isData: HomeModule.getHomeData(state),
    modal: HomeModule.getModal(state),
    transactions: HomeModule.getTransactions(state),
})

/*
*  los siguientes elementos
*           - getRequestCount: se hace el llamado al backend e ingresa los valores a isData
*           - getOnClose: se utiliza para cerrar un modal de error
*           - getOnOpen: se utiliza para abrir un modal de error
*           - onModal:  se utiliza para abrir  un modal de informaciÃ³n
*           - getclearMessage: limpieza de informacion
* */
const mapDispatchToProps = (dispatch) => ({
    getRequestCount: (request)=>{
        HomeModule.getCounts(dispatch)
    },
    getOnClose: (request) =>{
        HomeModule.onClose(dispatch)
    },
    getOnOpen: (request) =>{
        HomeModule.onOpen(dispatch)
    },
    onModal: (modal) =>{
        HomeModule.onModal(dispatch, modal)
    },
    onclearMessage:() => {
        HomeModule.clearMessageInfo(dispatch)
    },
})
export default connect (mapStateToProps, mapDispatchToProps)(HomeComponents)