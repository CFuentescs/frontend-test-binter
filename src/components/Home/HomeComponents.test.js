import React from 'react'
import { shallow } from '../../testing'
import HomeContainer from "./HomeComponents";
import toJson from "enzyme-to-json";
import '@testing-library/jest-dom/extend-expect';


const isData = [
    {
        "id": "3",
        "title": "Ya no viajo en moto",
        "ranking": [
            {
                "position": 1,
                "word": "moto",
                "ocurrences": "8"
            }],
    },
]

const homeMessageInfo = {
    open: false,
    title: '',
    message: '',
    suggestion: '',
    btnCloseLabel: '',
    warning: true,
    closable: true,
}

const isProcess = {
    open: false,
    title: '',
    message: '',
}

const modal = {
    open: false,
    title: '',
    message: '',
    suggestion: '',
    btnCloseLabel: '',
    warning: true,
    closable: true,
}
const error ={
    open: false,
    title: '',
    message: '',
}

const defaultProps = {
    isError: error,
    isProcess: isProcess,
    isLoading: false,
    isMessageInfo: Object.create(homeMessageInfo),
    isData: isData,
    getRequestCount:jest.fn(),
    getOnOpen: jest.fn(),
    getOnClose: jest.fn(),
    onClearMessageInfo: jest.fn(),
    onModal: jest.fn(),
    modal: modal,
}

describe('home Test', () => {
    it('renders correctly enzyme', () => {
        const wrapper = shallow(<HomeContainer {...defaultProps} />);
        let data = toJson(wrapper)
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders correctly', () => {
        const wrapper = shallow(<HomeContainer {...defaultProps} />);
        let data = wrapper.find('.homeComponent-view')
        expect(data).toHaveLength(1)
    });
})