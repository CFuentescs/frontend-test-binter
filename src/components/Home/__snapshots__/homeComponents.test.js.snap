// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home Test renders correctly 1`] = `
<Fragment>
  <div
    className="App homeComponent-view"
  >
    <br />
    <Container
      className="bg-light"
      fluid={true}
    >
      <Row>
        <Col>
          <h3>
             (id:
            )
          </h3>
        </Col>
        <Col>
          <Button
            onClick={[Function]}
            variant="primary"
          >
            Refrescar
          </Button>
        </Col>
      </Row>
      <Row>
        <Col>
          <div />
        </Col>
      </Row>
      <LoaderDialog
        message=""
        open={false}
        title=""
      />
    </Container>
  </div>
</Fragment>
`;

exports[`home Test renders correctly enzyme 1`] = `
<Fragment>
  <div
    className="App homeComponent-view"
  >
    <br />
    <Container
      className="bg-light"
      fluid={true}
    >
      <Row>
        <Col>
          <h3>
             (id:
            )
          </h3>
        </Col>
        <Col>
          <Button
            onClick={[Function]}
            variant="primary"
          >
            Refrescar
          </Button>
        </Col>
      </Row>
      <Row>
        <Col>
          <div />
        </Col>
      </Row>
      <LoaderDialog
        message=""
        open={false}
        title=""
      />
    </Container>
  </div>
</Fragment>
`;
